test/integration/collection-management/collection.test.ts

```diff
@@ -95,18 +95,14 @@ describe('Collection', function () {
       ]);
     });
 
-    it('should fail due to illegal listCollections', function (done) {
-      expect(() => db.collection(5)).to.throw('Collection name must be a String');
-      expect(() => db.collection('')).to.throw('Collection names cannot be empty');
-      expect(() => db.collection('te$t')).to.throw("Collection names must not contain '$'");
-      expect(() => db.collection('.test')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test.')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test..t')).to.throw('Collection names cannot be empty');
-      done();
+    it('fails on server due to invalid namespace', async function () {
+      const error = await db
+        .collection('a\x00b')
+        .insertOne({ a: 1 })
+        .catch(error => error);
+      expect(error['name']).to.equal('MongoServerError');
```

Comment:
```suggestion
      expect(error).to.have.property('name', 'MongoServerError');
```

Just another bit of chai assertion improvement advice. By checking for "to have prop" and passing the value as the second argument if the assertion fails it will print the whole error object, which provides a good reference for what might have gone wrong if the test fails.

Reply:
Also, I would go back to checking `instanceof` as it is the recommendation we give our users for filtering errors: https://github.com/mongodb/node-mongodb-native?tab=readme-ov-file#error-handling but I do think it would be great to keep the code number assertion as well.

Reply:
Sounds good, I'll push this in my next commit. I originally think ` instanceof` wasn't working because I didn't have my imports straight, but I've got it working fine now.